web/src/copy_and_paste.ts

```diff
@@ -661,15 +699,33 @@ export function paste_handler(this: HTMLTextAreaElement, event: JQuery.Triggered
 
         // Only intervene to generate formatted links when dealing
         // with a URL and a URL-safe range selection.
-        if (isUrl(trimmed_paste_text) && is_safe_url_paste_target($textarea)) {
-            event.preventDefault();
-            event.stopPropagation();
-            const url = trimmed_paste_text;
-            compose_ui.format_text($textarea, "linked", url);
-            return;
+        if (isUrl(trimmed_paste_text)) {
+            if (is_safe_url_paste_target($textarea)) {
+                event.preventDefault();
+                event.stopPropagation();
+                const url = trimmed_paste_text;
+                compose_ui.format_text($textarea, "linked", url);
+                return;
+            }
+            if (!compose_ui.cursor_inside_code_block($textarea) && !compose_ui.shift_pressed) {
```

Comment:
Put a blank line before this next `if` statement; it helps readability to leave some space between separate thoughts.

Reply:
Done